core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -2333,6 +2333,25 @@ wasm_set_running_mode(WASMModuleInstance *module_inst, RunningMode running_mode)
     return set_running_mode(module_inst, running_mode, false);
 }
 
+bool
+wasm_runtime_instantiate_run_start_func(wasm_module_inst_t module_inst,
+                                        wasm_module_inst_t parent,
+                                        wasm_exec_env_t exec_env,
+                                        char *error_buf,
+                                        uint32_t error_buf_size)
+{
+    const bool is_sub_inst = parent != NULL;
+    if (!execute_post_instantiate_functions((WASMModuleInstance *)module_inst,
+                                            is_sub_inst, exec_env)) {
+        set_error_buf(error_buf, error_buf_size,
+                      ((WASMModuleInstance *)module_inst)->cur_exception);
+        wasm_runtime_deinstantiate(module_inst);
```

Comment:
as wasm_runtime_instantiate succeeded at this point,
isn't it simpler to leave wasm_runtime_deinstantiate to api users?